internal/expr/parser.go

```diff
@@ -386,6 +392,42 @@ func (p *Parser) parseColumn() (fullName, bool, error) {
 	return fullName{}, false, nil
 }
 
+func (p *Parser) parseFuncName() (string, bool, error) {
+	cp := p.save()
+	if p.skipName() && p.skipByte('(') {
+		bracketLevel := 1
+		for {
+			if p.pos == len(p.input) {
+				return "", false, nil
+			}
+			if ok, err := p.skipStringLiteral(); err != nil {
+				return "", false, err
+			} else if ok {
+				continue
+			}
+			if ok := p.skipComment(); ok {
+				continue
+			}
+			if p.skipByte('(') {
+				bracketLevel++
+				continue
+			}
+			if p.skipByte(')') {
+				bracketLevel--
+				if bracketLevel == 0 {
+					break
+				}
+				continue
+			}
+			p.pos++
```

Comment:
It doesn't feel great that we're picking up what specific kind of syntax is supported here. This is very brittle and will fail often. We need to design this so that we don't need to be valiidating the details of this expression, just like we don't validate everywhere else except for the details that we do care about. So I'd try to consider that whole left side of the expression as either a column name (table, asterisk, etc), in which case we know exactly what's going on, or as a function holding arbitrary expressions as parameters, in which case we need to recurse into the parser. A good litmus test that this is well designed is supporting input parts inside the function parameters.

Reply:
I've changed `parseFunc` to recurse into the parser if we find a SQL function, it now works with nested input expressions.

The function `parseColumns` has the behaviour you describe above, it first checks for a function as a column, then for a regular column name